### ISSUE TITLE
Unable to Parse Ternary Operator Expression in Function Argument

### ISSUE THREAD

**Issue by morris821028** [beanshell/beanshell#568]
## Problem ##

```
_func() {
    a = new Integer(1);
    return String.format("%s" + "a" + "%s", a == null ? "3" : "4", a == 1 || a == null ? "5" : "6");
}

System.err.printf(_func());
```

## Env ##

* BeanShell 3.0.0-SNAPSHOT.841

## Expect Result ##

```
4a5
```

Reference https://github.com/javaparser/javaparser/blob/master/javaparser-core/src/main/javacc/java.jj#L1429

```java
package bsh;

import org.junit.Assert;
import org.junit.Test;


public class TernaryOperatorTest {
	static String _func() {
		Integer a = 1;
		return String.format("%s" + "a" + "%s", a == null ? "3" : "4",
				a == 1 || a == null ? "5" : "6");
	}

	static final String funcScript = "String _func() {\n" + "    Integer a = 1;\n"
			+ "    return String.format(\"%s\" + \"a\" + \"%s\", a == null ? \"3\" : \"4\", a == 1 || a == null ? \"5\" : \"6\");\n"
			+ "}";

	public static void main(String[] args) {
		System.err.println(funcScript);
		System.err.printf("%s\n", _func());
	}

	@Test
	public void parseInArgument() throws Exception {
		final Interpreter interpreter = new Interpreter();

		interpreter.eval(funcScript);
		String v = (String) interpreter.eval("_func()");
		Assert.assertEquals("4a5", v);
	}
}
```

**Comment by nickl-** [comment]
@morris821028  Well spotted!

This test can be simplified:

```java
print(true || false ? "True" : "False");
```

or even simpler still:

```java
res = true || false ? "True" : "False";
```

This is a parser error and needs to be fixed in grammar.

As a workaround placing the condition in brackets works.

```java
print((true || false) ? "True" : "False");
``` 

But since this works in kava it should also work the same in BeanShell.

**Comment by nickl-** [comment]
This may be a simpler root cause for the issue raised at #390

**Comment by opeongo** [comment]
This broke due to the removal of operator precedence in 8e712ba, and is fixed in PR #671

**Comment by nickl-** [comment]
Closed works as expected
